web/app/src/components/navigation/NavBar.js

```diff
@@ -14,6 +14,21 @@ const NavBar = () => {
         textShadow:'0 0.015em #177368,0 -0.015em #177368,0.01em 0 #177368,-0.01em 0 #177368',
     };
 
+    const [name, setName] = useState('');
+    const [firstname, setFirstname] = useState('');
+    const [sciper, setSciper] = useState(0);
+
+    fetch('/api/getpersonnalinfo')
```

Comment:
you're already doing fetching the personal info in `App.js` and you should do it only once. When you do it in `App.js` there you could save all the personal info in a state and use it there instead of doing another call.